### DISCUSSION TITLE
Connection String not working

### DISCUSSION THREAD

**Discussion by DuffCorp** [t3-oss/create-t3-turbo#1131] [Q&A]
Hi,

Im trying to initialise my first repo with T3 Turbo as its perfect for a project I need to do, I cloned the repo and filled out the .env with the relevant details, However when trying to start the development servers with

`pnpm dev`

with the following connection string

`POSTGRES_URL="postgresql://postgres.zwogfdvyhrhatnkgimafaafeasfaffwafazvn:examplepassword@aws-0-eu-west-2.pooler.supabase.com:6543/postgres"`

I get the following error

`Tasks                │ @acme/nextjs#dev > cache bypass, force executing d1773eb255b34811
─────────────────────│> dotenv -e ../../.env -- "next" "dev"
@acme/nextjs#dev    »│
@acme/expo#dev      »│  ▲ Next.js 14.2.4
@acme/validators#dev✓│  - Local:        http://localhost:3000
@acme/db#dev        ✓│
@acme/ui#dev        ✓│ ✓ Starting...
@acme/api#dev       ✓│ ✓ Ready in 2.4s
                     │ ○ Compiling /src/middleware ...
                     │ ✓ Compiled /src/middleware in 760ms (467 modules)
                     │ ○ Compiling / ...
                     │ ✓ Compiled / in 1288ms (2498 modules)
                     │>>> tRPC Request from rsc by undefined
                     │ ⨯ Internal error: TypeError: Cannot read properties of undefined (reading 'default')
                     │    at resolveClientReference (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:87:55)
                     │    at resolveModule (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1667:35)
                     │    at processFullRow (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1740:25)
                     │    at processBinaryChunk (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1858:21)
                     │    at progress (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1926:17)
                     │digest: "1732920403"
                     │[TypeError: Cannot read properties of undefined (reading 'default')]
                     │ ⨯ [TypeError: Cannot read properties of undefined (reading 'default')]
                     │ ⨯ [TypeError: Cannot read properties of null (reading 'default')]
                     │ GET / 500 in 1723ms
                     │[TRPC] post.all took 417ms to execute
                     │A query that was dehydrated as pending ended up rejecting. [[["post","all"],{"type":"query"}]]: TRPCError: VercelPostgresError - 'invalid_connection_string': This connection string is meant to be 
                     │sed with a direct con
                     │nection. Make sure to use a pooled connection string or try `createClient()` instead.; The error will be redacted in production builds
                     │ ⨯ Internal error: Error: redacted
─────────────────────│    at eval (../../node_modules/@tanstack/query-core/build/modern/hydration.js:37:31)`

Feel like I'm loosing my mind trying to find the error as the connection string is fine. I can even still use the push command and that works fine but for some reason I just cant start a development server to start working on my project.

if I use the following structure connection string I get the following error

`POSTGRES_URL="postgresql://postgres.zwogfdvyhrhatnkgimafaafeasfaffwafazvn:examplepassword@aws-0-eu-west-2.pooler.supabase.com:6543/postgres?workaround=supabase-pooler.vercel"`

`Tasks                │ @acme/nextjs#dev > cache bypass, force executing 4276b8aa2448280a
─────────────────────│> pnpm with-env next dev
@acme/nextjs#dev    »│
@acme/expo#dev      »│
@acme/validators#dev✓│> @acme/nextjs@0.1.0 with-env C:\Users\dd1du\Documents\Projects\CRM_V2\apps\nextjs
@acme/db#dev        ✓│> dotenv -e ../../.env -- "next" "dev"
@acme/ui#dev        ✓│
@acme/api#dev       ✓│  ▲ Next.js 14.2.4
                     │  - Local:        http://localhost:3000
                     │
                     │ ✓ Starting...
                     │ ✓ Ready in 2.5s
                     │ ○ Compiling /src/middleware ...
                     │ ✓ Compiled /src/middleware in 811ms (467 modules)
                     │ ○ Compiling / ...
                     │ ✓ Compiled / in 1260ms (2498 modules)
                     │>>> tRPC Request from rsc by undefined
                     │ ⨯ Internal error: TypeError: Cannot read properties of undefined (reading 'default')
                     │    at resolveClientReference (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:87:55)
                     │    at resolveModule (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1667:35)       
                     │    at processFullRow (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1740:25)      
                     │    at processBinaryChunk (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1858:21)  
                     │    at progress (../../node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-client.edge.development.js:1926:17)
                     │digest: "1732920403"
                     │[TypeError: Cannot read properties of undefined (reading 'default')]
                     │ ⨯ [TypeError: Cannot read properties of undefined (reading 'default')]
                     │ ⨯ [TypeError: Cannot read properties of null (reading 'default')]
                     │ GET / 500 in 1750ms
                     │[TRPC] post.all took 866ms to execute`

**Comment by artem-alek** [comment]
You are probably missing the `?workaround=supabase-pooler.vercel` at the end of the url add that and see if it works
